sway-core/src/semantic_analysis/namespace/module.rs

```diff
@@ -400,6 +403,183 @@ impl Module {
 
         ok((), warnings, errors)
     }
+
+    /// Pull a single variant `variant` from the enum `enum_name` from the given `src` module and import it into the `dst` module.
+    ///
+    /// Paths are assumed to be relative to `self`.
+    pub(crate) fn variant_import(
+        &mut self,
+        engines: Engines<'_>,
+        src: &Path,
+        enum_name: &Ident,
+        variant_name: &Ident,
+        dst: &Path,
+        alias: Option<Ident>,
+    ) -> CompileResult<()> {
+        let mut warnings = vec![];
+        let mut errors = vec![];
+
+        let decl_engine = engines.de();
+
+        let src_ns = check!(
+            self.check_submodule(src),
+            return err(warnings, errors),
+            warnings,
+            errors
+        );
+        match src_ns.symbols.get(enum_name).cloned() {
```

Comment:
Similarly here, for the rest of this match statement.  It's often more concise to use `map()`, `and_then()`, `unwrap_or_else()`, etc. for `Option` and `Result` rather than matching `Some` or `None`.